controlplane/kubeadm/internal/cluster.go

```diff
@@ -107,7 +107,7 @@ func (m *Management) GetWorkloadCluster(ctx context.Context, clusterKey client.O
 		RootCAs:      caPool,
 		Certificates: []tls.Certificate{clientCert},
 	}
-
+	cfg.InsecureSkipVerify = true
```

Comment:
This is being done because the endpoint used to be statically set to "127.0.0.1" which was in the SAN list for the etcd certs. With these changes, the TLS verify was failing silently as we now use a dynamic set of endpoints.

Reply:
Would it be possible to get an example where the endpoint advertised by the cluster isn't included in the SAN list for the etcd certs? kubeadm should be generating the certificates to include both localhost and the host IP in the listed SANs.

Reply:
Kubeadm is doing the right thing but because we have our own dialer, we don't use endpoints as they were meant to be used. Right now, we send the etcd pod names as endpoints which will of course not be included in the SAN list. Our dialer then uses the podname to figure out which pod to set up the port forwarding to.

Reply:
Yup, what @gab-satchi said. The address that is being attempted to be verified is `pod.namespace`, API server itself is providing some guarantee of routing us to the correct place and mutual TLS authentication means we don't really need to verify the identity further.